vignettes/tutorial_geophysics.Rmd

```diff
@@ -0,0 +1,149 @@
+print(head(d, 30))
+
+# Create a scatter plot
+plot(d$X, d$Y, pch = 20, main = 'Svalbard MT sites', xlab = 'X', ylab = 'Y')
+
+```
+
+```{r}
+locations = d[, c('Lon', 'Lat')]
+locations = unique(locations)
+names(locations)=c('LONG','LAT')
+```
+
+## Meshing
+```{r}
+
+mesh <- fmesher::fm_mesh_2d(loc.domain = locations,
+                                max.edge = 0.05,
+                                cutoff = 1e-3,
+                                offset=0.1
+                                )
+plot(mesh)
+points(locations, col = "red")
+
+fdmr::plot_mesh(mesh)
+
+```
+
+## Stochastic modeling
+```{r}
+library(INLA)
+
+# Synthetic data generation
+set.seed(123) # For reproducibility
+n <- 100 # Number of time points
+stations <- 12 # Number of stations
+
+# Generate time index
+time_index <- 1:n
+
+# Generate spatial index (station IDs)
+space_index <- rep(1:stations, each = n)
+
+# Simulate some harmonic signals with noise for three stations
+harmonic_data <- data.frame(
+  station = factor(space_index),
+  time = rep(time_index, stations),
+  observation = sin(rep(time_index, stations) * 2 * pi / 50) + 
+                rnorm(n * stations, sd = 0.5) +
+                rep(rnorm(stations, sd = 3), each = n) # Station-specific offset
+)
+
+# Define the model with harmonic terms for time and spatial correlation
+formula <- observation ~ f(station, model = "iid") +
+  f(time, model = "rw1", cyclic = TRUE)
+
+# Fit the model using INLA
+result <- inla(formula, family = "gaussian", data = harmonic_data)
+
+# Display the summary of the results
+summary(result)
+
+# Visualize the fitted values
+plot(harmonic_data$time, harmonic_data$observation, col = harmonic_data$station, pch = 19, cex = 0.5, xlab = "Time", ylab = "Observation", main = "MT signals at 12 stations")
+points(harmonic_data$time, result$summary.fitted.values$mean, pch = 4, cex = 0.7, col = "blue")
+legend("topright", legend = c("Observations", "Fitted"), col = c("black", "blue"), pch = c(19, 4))
+
+
+# Extract the hyperparameters of the spatial field
+spatial_hyperparams <- result$summary.hyperpar
+
+# Print the hyperparameters
+print(spatial_hyperparams)
+
+```
+## Observed time series will come here.. 
+
+
+## This part doesn't work
+
+```{r}
+simdf <- data.frame(
+  ID = d$Code,
+  time = d$T,
+  datn = Re(d$Zxx),
+  E = Re(d$DZxx),
+  LONG = d$Lon,
+  LAT = d$Lat
+)
+
+
+initial_range = 0.1
+spde <- INLA::inla.spde2.pcmatern(
+  mesh = mesh,
+  prior.range = c(initial_range, 0.5),
+  prior.sigma = c(1, 0.01)
+)
+
+sigma0 <- 100
+range0 <- 0.03
+kappa0 <- sqrt(8 / 1) / range0
+tau0 <- 1 / (sqrt(4 * pi) * kappa0 * sigma0)
+
+inla.seed <- sample.int(n = 1E4, size = length(d$T))
+Q <- INLA::inla.spde.precision(spde, theta = c(log(tau0), log(kappa0)))
+x.mat <- matrix(NA, ncol = length(d$T), nrow = mesh$n)
+for (co in 1:ncol(x.mat)) {
+  x.mat[, co] <- INLA::inla.qsample(n = 1, Q)
+}
+
+A <- INLA::inla.spde.make.A(mesh = mesh, loc = locations)
```

Comment:
the package down is failing because of this line, but i dont understand what is wrong with the mesh/locations @gareth-j do you know?